src/components/UserSettingsForm/UserSettingsForm.jsx

```diff
@@ -1,5 +1,227 @@
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import css from './UserSettingsForm.module.css';
+import * as Yup from 'yup';
+import { useState } from 'react';
+
+const userSettingsValidationSchema = Yup.object().shape({
+  userImage: Yup.mixed(),
+  gender: Yup.string().required('Please select your gender'),
+  userName: Yup.string()
+    .min(3, 'The minimum number of characters is 3')
+    .max(50, 'The maximum number of characters is 50')
+    .required('Name is required'),
+  userEmail: Yup.string().email('Invalid email format'),
+  userWeight: Yup.number(),
+  userTime: Yup.number(),
+  UserWaterNorma: Yup.number(),
+});
+
 const UserSettingsForm = () => {
-  return <div>UserSettingsForm</div>;
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    resolver: yupResolver(userSettingsValidationSchema),
+  });
+
+  const [imagePreview, setImagePreview] = useState();
+
+  const handleImageChange = event => {
+    const file = event.target.files[0];
+    if (file) {
```

Comment:
Please add case to clear old URL
if (file) {
    if (imagePreview) {
      URL.revokeObjectURL(imagePreview);  // clear old url 
    }
    setImagePreview(URL.createObjectURL(file));
  }